AgileCoffeeShop/Program.cs

```diff
@@ -0,0 +1,130 @@
+        private static List<CoffeeShop> coffeeShops;
+        private static Dictionary<string, double> distancesDictionary;
+        #endregion
+  
+        private static void ReadDataFromUrlFile(string urlFilePath)
+        {
+            WebClient client = new WebClient();
+            try
+            {
+                Stream stream = client.OpenRead(urlFilePath);
+                StreamReader reader = new StreamReader(stream);
+                string line = string.Empty;
+
+                while (!string.IsNullOrEmpty(line = reader.ReadLine()))
+                {
+                    string[] splitLine = line.Split(',');
+                    CoffeeShop coffeeShop = new CoffeeShop();
+                    coffeeShop.Name = splitLine[0];
+                    ValidateCoordinate(splitLine[1]);
+                    coffeeShop.Coordinate.YCoordinate = Convert.ToDouble(splitLine[1]);
+                    ValidateCoordinate(splitLine[2]);
+                    coffeeShop.Coordinate.XCoordinate = Convert.ToDouble(splitLine[2]);
+                    coffeeShops.Add(coffeeShop);
+                }
+            }
+            catch(Exception exception)
+            {
+                Console.WriteLine("Invalid URL." + exception.Message + "Program will exit.\n Press any key to continue...");
+                Console.ReadKey();
+            }
+        }
+
+        public static void ValidateCoordinate (string coordinate)
+        {
+            double temp = 0;
+            try
+            {
+                temp = Convert.ToDouble(coordinate);
+            }
+            catch(Exception exception)
+            {
+                Console.WriteLine("Invalid coordinates." + exception.Message + "Program will exit.\nPress any key to continue...");
+                Console.ReadKey();
+            }
+        }
+        private static void ValidateUrl(string url) 
+        {
+            bool result = Uri.TryCreate(url, UriKind.Absolute, out Uri uriResult) && (uriResult.Scheme == Uri.UriSchemeHttp || uriResult.Scheme == Uri.UriSchemeHttps);
+            if(result==true)
+            {
+                if(!url.Contains("raw.githubusercontent.com"))
+                {
+                    result = false;
+                }
+            }
+            if(result==false)
+            {
+                Console.WriteLine("Invalid URL. Program will exit.\n Press any key to continue...");
+                Console.ReadKey();
+            }
+        }
+
+        private static void Init()
+        {
+            user = new User();
+            urlFilePath = string.Empty;
+            coffeeShops = new List<CoffeeShop>();
+            distancesDictionary = new Dictionary<string, double>();
+        }
+
+        public static double CalculateDistance(Coordinate A, Coordinate B)
+        {
+            var distance = Math.Round(Math.Sqrt(Math.Pow((A.XCoordinate - B.XCoordinate), 2) + Math.Pow((A.YCoordinate - B.YCoordinate), 2)), 4);
```

Comment:
This is really hard to be read. Most of the time we are writing code for humans so making it more easy to read will help everybody on the long run. 
Could you refactor this to be more easy readable?

Reply:
Yes, I agree that is hard to be read and I will refactor. I will break it in several pieces.